3D Chess/Assets/Scripts/GameState.cs

```diff
@@ -16,15 +16,15 @@ public void MovePiece(Vector3 oldPosition, Vector3 newPosition)
     {
         realm.Write(() =>
         {
-            // Check if there is already a piece at the new position and if so, remove it.
+            // Check if there is already another piece at the new position and if so, remove it.
             var attackedPiece = FindPieceEntity(newPosition);
-
-            // Now move the piece to it's new position.
-            var movedPieceEntity = FindPieceEntity(oldPosition);
             if (attackedPiece != null)
             {
                 realm.Remove(attackedPiece);
             }
+
+            // Now move the active piece to it's new position.
+            var movedPieceEntity = FindPieceEntity(oldPosition);
```

Comment:
This might be a bit silly, but should we handle the case where there might not be a piece in the old position anymore? Unlikely to happen in the local case, but in the sync case, it's possible that you're moving a piece that your opponent destroys at the same time. If we want to be extra safe, we can probably abort the write transaction if we find that `movedPieceEntity` is null.

Reply:
Since chess is turned based this cannot happen. I'd neglect this under the 'rules are not implemented and have to be taken care by the players' umbrella. It's basically the same way we don't enforce pieces only moving how they are allowed to.

I'd love to add all those to have a proper example (I can think of some cool stuff we could do with functions there - I assume) but I don't know if that's over the top?

Reply:
Makes sense, let's not worry about it then